src/Juvix/Compiler/Core/Extra/Utils.hs

```diff
@@ -38,6 +40,9 @@ substEnvInBranch env br = over caseBranchBody (substEnv env') br
 isClosed :: Node -> Bool
 isClosed = not . has freeVars
 
+mkAxiom :: Interval -> Type -> Node
+mkAxiom loc ty = mkBottom (Info.setInfoLocation loc (Info.setInfoType ty mempty))
```

Comment:
I think the general question is whether we want those bottoms to always have types or not. If they are intrinsically typed (they can always have type Dynamic), then the type should be a separate field, not in the info, much like it is a separate field in `Binder`.

Reply:
In any case, storing Nodes in Infos can only be temporary, because the recursors don't descend into stuff stored in Info (by design). The type infos are used basically only by ComputeTypeInfo, but you need to run it just before using it, without some intervening transformations that could change the types (especially binders in them).

Reply:
I'll add a field

Reply:
done